azurerm/resource_arm_kubernetes_cluster.go

```diff
@@ -454,3 +539,73 @@ func resourceAzureRMKubernetesClusterServicePrincipalProfileHash(v interface{})
 
 	return hashcode.String(buf.String())
 }
+
+func base64Decode(str string) (string, bool) {
```

Comment:
not a huge thing, but would be nice to throw some unit testing on this and/or `getKubeConfig`

Reply:
I don't think we need this function, fwiw - we could just check that `err != nil` below?